src/main.tsx

```diff
@@ -1,27 +1,40 @@
 import React from "react";
 import { createRoot } from "react-dom/client";
-import { createHashRouter, RouterProvider } from "react-router-dom";
+import { createBrowserRouter, RouterProvider } from "react-router-dom";
 
 import { ErrorPage, LandingPage } from "./routes";
+import { decodeGitHubPagesUrl, isEncodedPathUrl, tryRemoveHashRouting } from "./utils/gh_routing";
 
 import AppStyle from "./components/AppStyle";
 import Viewer from "./Viewer";
 
+// Decode URL path if it was encoded for GitHub pages or uses hash routing.
+const locationUrl = new URL(window.location.toString());
+if (locationUrl.hash !== "" || isEncodedPathUrl(locationUrl)) {
+  const decodedUrl = tryRemoveHashRouting(decodeGitHubPagesUrl(locationUrl));
+  const newRelativePath = decodedUrl.pathname + decodedUrl.search + decodedUrl.hash;
+  console.log("Redirecting to " + newRelativePath);
+  // Replaces the query string path with the original path now that the
+  // single-page app has loaded. This lets routing work as normal below.
+  window.history.replaceState(null, "", newRelativePath);
+}
+
```

Comment:
Compare to https://github.com/allen-cell-animated/website-3d-cell-viewer/blob/main/public/index.tsx#L22-L31